### ISSUE TITLE
Multiple installed version handling

### ISSUE THREAD

**Issue by Beep6581** [Beep6581/RawTherapee#1319]
Originally reported on Google Code with ID 1335

```
Pursuing the discussion from this forum's post ( http://rawtherapee.com/forum/viewtopic.php?p=28144#p28144
), i think that we should solve some compatibility problem when multiple versions are
installed, as we recomend to keep previous ones for compatibility reasons.

The problem: if you mistakenly open a previous version, you may suppress newly introduced
parameters from the pp3 files. That's a serious security risk and i guess, unaceptable
by the user. Do we have to / can we take actions to avoid this?

I think that the bare minimum would be to alert users by two ways:
1. update the minor version number (i.e. yy in "xx.yy.zz") for each compatibility break
only (together with the pp3's version), and tell it in the release notes file. Incidentaly,
it will install RT in a separate directory, as well as create a separate cache. The
problem still remain for sidecar pp3 files!
2. Enhance RT to make post-install checks: each time RT read an older or newer version
pp3 file (based on the PP3 version string or AppVersion string), then a popup alert
saying "This processing file has been produced by a version that holds parameters differently
and produce different output. Click OK to convert it to this version, Cancel to abort
loading", with a "Don't ask again" option. This should let users take acions on what
to do.

Actual versions don't make this check, so there's a data loss risk if they make the
mistake explained above. To prevent this, the only possible solution that comes to
my mind is to make the newer pp3 files unreadable by actual RT version, and eventually
make RT crash when it tries to load PP3 files produced by newer versions.

To sum up, backward compatibility will be possible if accepted by the user, but upward
compatibility should be forbiden, for security reason.

Of course, this security is only possible if we rigorously increment the minor version
on compatibility break. We have to do it when the EPD tool will be updated as well
as the denoise tool.
```

Reported by `natureh.510` on 2012-04-21 12:07:06
